### PR TITLE
Feature/937 feature/933 updating users and permissions for a role

### PR DESCRIPTION
**Pull request by 121watts**
- [ ] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass

Connect #937 
Connect #933

### Note 
This currently only supports the updating of global permissions i.e. permissions not scoped to a specific database.

### CODE REVIEW

File: ui/src/admin/apis/index.js [hunk 1]
```diff
@@ -61,3 +61,31 @@ export const deleteUser = async (url, addFlashMessage, username) => {
     })
   }
 }
+
+export const updateRoleUsers = async (url, users) => {
```
Comment (jademcgough) [inline ui/src/admin/apis/index.js]: this and `updateRolePermissions` can be merged into a single function

Comment (121watts) [inline ui/src/admin/apis/index.js]: I was considering that.  Ultimately, I didn't want to do logic around what to do if a particular property wasn't present.  Maybe I'm missing something?

File: ui/src/admin/components/RoleRow.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import _ from 'lodash'
 import MultiSelectDropdown from 'shared/components/MultiSelectDropdown'
 import DeleteRow from 'src/admin/components/DeleteRow'
 
-const PERMISSIONS = [
+const ALL_PERMISSIONS = [
```
Comment (jaredscheib) [inline ui/src/admin/components/RoleRow.js]: add a TODO here until we get this from server

File: ui/src/admin/actions/index.js [hunk 3]
```diff
@@ -104,3 +108,21 @@ export const deleteUserAsync = (user, addFlashMessage) => (dispatch) => {
   // delete user on server
   deleteUserAJAX(user.links.self, addFlashMessage, user.name)
 }
+
+export const updateRoleUsersAsync = (users, role) => async (dispatch) => {
```
Comment (jaredscheib) [inline ui/src/admin/actions/index.js]: switch order of arguments to follow convention in the rest of codebase. and actually, the other convention is just to pass in the url directly. let's choose one or the other and stick with it.

File: ui/src/admin/actions/index.js [hunk 4]
```diff
@@ -104,3 +108,21 @@ export const deleteUserAsync = (user, addFlashMessage) => (dispatch) => {
...
+  try {
+    await updateRoleUsersAJAX(role.links.self, users)
+    dispatch(publishNotification('success', 'Role users updated'))
+  } catch (error) {
+    dispatch(publishNotification('error', `Failed to update role: ${error.data.message}`))
+  }
+}
+
+export const updateRolePermissionsAsync = (permissions, role) => async (dispatch) => {
```
Comment (jaredscheib) [inline ui/src/admin/actions/index.js]: switch order of args

**Comment by jaredscheib** [comment]
Also, @121watts-- update the changelog

**Pull request merged**